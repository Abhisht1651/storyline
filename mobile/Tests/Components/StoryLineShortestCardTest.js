// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import StoryLineShortestCard from '../../App/Components/StoryLineShortesCard'

describe('StoryLineShortestCard type 1', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineShortestCard />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineShortestCard />)
    expect(wrapper.find('Text')).toHaveLength(3)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineShortestCard />)
    expect(wrapper.find('Image')).toHaveLength(2)
  })
})

describe('StoryLineShortestCard type 2', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineShortestCard type={2} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineShortestCard type={2} />)
    expect(wrapper.find('Text')).toHaveLength(3)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineShortestCard type={2} />)
    expect(wrapper.find('Image')).toHaveLength(1)
  })
})

describe('StoryLineShortestCard type 3', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineShortestCard type={3} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineShortestCard type={3} />)
    expect(wrapper.find('Text')).toHaveLength(5)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineShortestCard type={3} />)
    expect(wrapper.find('Image')).toHaveLength(2)
  })
})
