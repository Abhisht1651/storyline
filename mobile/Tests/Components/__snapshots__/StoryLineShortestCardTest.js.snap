// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryLineShortestCard type 1 should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 136,
          "paddingRight": 16,
          "width": 220,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.08,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          World
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#1F1F60",
              "borderRadius": 2,
              "height": 4,
              "marginHorizontal": 10,
              "width": 4,
            }
          }
        />
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "width": 34,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.08,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
           Trending
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingTop": 4,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 32,
              "letterSpacing": -0.11,
              "lineHeight": 44,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Ray Contreras talks about the different accents
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`StoryLineShortestCard type 2 should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 136,
          "paddingRight": 16,
          "width": 220,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingTop": 4,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 32,
              "letterSpacing": -0.11,
              "lineHeight": 44,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Ray Contreras talks about the different accents
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "textAlign": "left",
            }
          }
        >
          Source by
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "paddingHorizontal": 5,
              "textAlign": "left",
            }
          }
        >
          Forbes
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`StoryLineShortestCard type 3 should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.08,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        World
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#1F1F60",
            "borderRadius": 2,
            "height": 4,
            "marginHorizontal": 10,
            "width": 4,
          }
        }
      />
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 34,
            "width": 34,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.08,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
         Trending
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 32,
          "letterSpacing": -0.11,
          "lineHeight": 44,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingTop": 11.5,
          "textAlign": "left",
        }
      }
    >
      UK exit from the European Union
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 136,
          "paddingRight": 16,
          "width": 220,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28,
              "letterSpacing": 0,
              "marginTop": 7.5,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Sept 21, 2019
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingTop": 4,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 32,
              "letterSpacing": -0.11,
              "lineHeight": 44,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Ray Contreras talks about the different accents
        </Text>
      </View>
    </View>
  </View>
</View>
`;
